scalding-commons-macros/src/test/scala/com/twitter/scalding/commons/macros/PlatformTest.scala

```diff
@@ -87,32 +90,80 @@ class PlatformTest extends WordSpec with Matchers with HadoopSharedPlatformTest
   }
 
   "TBase Test" should {
-    "Expected items should match: TestThriftStructure" in {
+    "Expected items should match: TestThriftStructure2" in {
       runCompareTest[TestThriftStructure]
     }
   }
 
   "ThriftStruct Test" should {
 
-    "Expected items should match : TestStruct" in {
-      runCompareTest[TestStruct]
+    "Expected items should match : Internal Serializer / TestStructdd" in {
+
+      runCompareTest[TestStruct](toScroogeInternalOrderedSerialization[TestStruct], implicitly)
+    }
+
+    "Expected items should match : TProtocol / TestStruct" in {
+      implicit def toScroogeInternalOrderedSerialization[T]: OrderedSerialization[T] = macro ScroogeInternalOrderedSerializationImpl[T]
```

Comment:
By the way, your comment is good here, but it would be better to also add it as a comment in the code to help the next reader who may not be you are I.